web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -45,43 +54,54 @@ public static function create(ContainerInterface $container, array $configuratio
       $plugin_definition,
       $container->get('config.factory'),
       $container->get('logger.factory'),
+      $container->get('database'),
       $container->get('http_client_factory'),
+      $container->get('weather_block.get_set_city_user'),
+      $container->get('cache.default'),
+      $container->get('current_user'),
+      $container->get('weather_block.fetch_api_data'),
     );
   }
 
   /**
    * {@inheritdoc}
    */
   public function build(): array {
-    $user = \Drupal::currentUser();
-    $user_id = $user->id();
-
-    $user_config = $this->configuration['users'][$user_id] ?? [];
-    $cities = $user_config['cities'] ?? [];
-    $api_key = $this->configuration['settings']['key'] ?? '';
-
-    if (!is_array($cities)) {
-      $cities = [];
-    }
+    $user_id = $this->accountProxy->id();
+    $cityArray = $this->cityService->getSavedCityForUser($user_id);
+    $city = implode(', ', $cityArray);
```

Comment:
Again, it is very unclear what is happening here.
The method is called saved CITY for the user.
But your variable cityArray
What array? Is there something more than only the name of the city? 
Is it multiple cities?

If this logic does make sense, leave a comment explaining what is happening here.